modules/dnn/src/layers/dft_layer.cpp

```diff
@@ -0,0 +1,296 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+// Copyright (C) 2025, BigVision LLC, all rights reserved.
+// Third party copyrights are property of their respective owners.
+
+#include "../precomp.hpp"
+#include <opencv2/dnn/shape_utils.hpp>
+#include <opencv2/core.hpp>
+#include <numeric>
+
+// ONNX DFT operator
+// Spec: https://onnx.ai/onnx/operators/onnx__DFT.html
+// Supported opset: 17-22
+
+namespace cv {
+namespace dnn {
+
+template<typename Scalar, typename ComplexVec, typename FillFn>
+static void dftAlongAxisWorker(const Mat& src,
+                               Mat& dst,
+                               const std::vector<int>& dimSizesSrc,
+                               const std::vector<size_t>& stridesSrc,
+                               const std::vector<size_t>& stridesDst,
+                               const std::vector<int>& iterDims,
+                               const std::vector<int>& outerSizes,
+                               const std::vector<size_t>& outerStep,
+                               const size_t totalOuter,
+                               const int axis,
+                               const int N,
+                               const size_t strideAxisSrc,
+                               const size_t strideAxisDst,
+                               const int inMatType,
```

Comment:
why is this passed as function parameter when you can extract it from `Mat& src`?